### PR TITLE
Darkmode db

### PR DESCRIPTION
**Pull request by jessywork**

### CODE REVIEW

File: GrocifyApp.API/Controllers/GenericController.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace GrocifyApp.API.Controllers
 
     [ApiController]
     [Route("api/[controller]")]
-    [Authorize]
+    //[Authorize]
```
Comment (Catarina98) [inline GrocifyApp.API/Controllers/GenericController.cs]: remove comment

File: GrocifyApp.UI/src/pages/Settings.jsx [hunk 2]
```diff
@@ -61,18 +61,53 @@ function Settings(props) {
...
     const [searchInput, setSearchInput] = useState('');
-    const [darkMode, setDarkMode] = useState(false);
+    const [darkMode, setDarkMode] = useState(props.userAuth ? props.userAuth.isDarkMode : false);
+    const [errorMessage, setErrorMessage] = useState('');
     const navigate = useNavigate();
 
+    const updateUserDarkMode = async () => {
+        
+        try {
+            var p = props.userAuth.password; //change
```
Comment (Catarina98) [inline GrocifyApp.UI/src/pages/Settings.jsx]: User is a special entity, so we will not use the Update endpoint from Generic. Please create a new enpoint: "ToggleDarkMode". You don't need to pass information from user. You just need to pass the token in Header, and in the API you will have access to the authenticated user.

File: GrocifyApp.API/Controllers/AuthController.cs [hunk 3]
```diff
@@ -104,7 +104,7 @@ public async Task<ActionResult<string>> Login([FromBody] LoginRequestModel user)
                 PasswordSalt = getUser.PasswordSalt,
             };
 
-            User = new UserResponseModel() { Id = getUser.Id, Email = getUser.Email, Name = getUser.Name, HouseId = houseId };
+            User = new UserResponseModel() { Id = getUser.Id, Email = getUser.Email, Name = getUser.Name, HouseId = houseId, Password = getUser.Password, ConfirmPassword = getUser.Password };
```
Comment (Catarina98) [inline GrocifyApp.API/Controllers/AuthController.cs]: revert this changes, never return the users passwords!

File: GrocifyApp.API/Models/ResponseModels/UserResponseModel.cs [hunk 4]
```diff
@@ -11,6 +11,15 @@ public class UserResponseModel
         /// <example>User Email</example>
         public required string Email { get; set; }
 
+        /// <example>Password</example>
```
Comment (Catarina98) [inline GrocifyApp.API/Models/ResponseModels/UserResponseModel.cs]: Remove Password and ConfirmPassword

File: GrocifyApp.UI/src/pages/Settings.jsx (hunk 2 above)
Comment (Catarina98) [reply GrocifyApp.UI/src/pages/Settings.jsx]: To pass Token:

if (localStorage.getItem('token') == undefined) {
return;
}

const response = await fetch(api/User/ToggleDarkMode, {
method: 'PUT',
headers: {
'Content-Type': 'application/json',
'Authorization': Bearer ${localStorage.getItem('token')},
},
});

File: GrocifyApp.UI/src/App.jsx [hunk 5]
```diff
@@ -25,6 +27,36 @@ PrivateRoute.propTypes = {
...
+    const userId = "581ccc32-dd5d-455b-d2c2-08dc11ed02ad";
+    const [errorMessage, setErrorMessage] = useState('');
+
+    useEffect(() => {
+        getUserDarkMode();
+    }, []);
+
+    const getUserDarkMode = async () => {
+        try {
+            const response = await fetch(`api/User/${userId}`, {
```
Comment (Catarina98) [inline GrocifyApp.UI/src/App.jsx]: remove this call to API. when you do the login, you have the information of the user, with DarkMode. So, you can save darkmode in localStorage, as you do for token. And have no need to always make this call at the beggining, what do you think?

File: GrocifyApp.UI/src/pages/Settings.jsx (hunk 2 above)
Comment (Catarina98) [reply GrocifyApp.UI/src/pages/Settings.jsx]: if you followed my suggestion of saving darkMode in localStorage, don't forget to update it's value here

**Review by Catarina98 — CHANGES_REQUESTED** [review]

File: GrocifyApp.API/Controllers/AuthController.cs [hunk 6]
```diff
@@ -99,23 +99,32 @@ public async Task<ActionResult<string>> Login([FromBody] LoginRequestModel user)
...
+
+            return Ok(loginResponseModel);
         }
 
         private string CreateToken(UserRequestModel userAPI)
         {
             List<Claim> claims = new List<Claim>
             {
-                new Claim(ClaimTypes.Name, userAPI.Name)
+                new Claim(ClaimTypes.Name, userAPI.Name),
```
Comment (Catarina98) [inline GrocifyApp.API/Controllers/AuthController.cs:118]: why Claims Email?

File: GrocifyApp.API/Controllers/UserController.cs [hunk 7]
```diff
@@ -35,5 +36,44 @@ public async Task<ActionResult<User>> GetUserByEmail(string email)
 
             return Ok(getEntity);
         }
+
+        [HttpPut("toggleDarkMode")]
+        public async Task<ActionResult<User>> ToggleDarKMode()
+        {
+            var emailClaim = User.FindFirst(ClaimTypes.Email);
```
Comment (Catarina98) [inline GrocifyApp.API/Controllers/UserController.cs]: use Authenticator.User

File: GrocifyApp.API/Controllers/AuthController.cs [hunk 8]
```diff
@@ -99,23 +99,32 @@ public async Task<ActionResult<string>> Login([FromBody] LoginRequestModel user)
             {
```
Comment (Catarina98) [inline GrocifyApp.API/Controllers/AuthController.cs]: is it possible to remove userRequestModel and use User instead?

File: GrocifyApp.API/Models/RequestModels/UserRequestModel.cs [hunk 9]
```diff
@@ -13,6 +13,9 @@ public class UserRequestModel
         [RegularExpression(@"^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$", ErrorMessage = GenericConsts.RequestModels.ValidEmail)]
         public required string Email { get; set; }
 
+        /// <example>false</example>
```
Comment (Catarina98) [inline GrocifyApp.API/Models/RequestModels/UserRequestModel.cs:16]: remove?

File: GrocifyApp.UI/src/consts/ApiEndpoints.js [hunk 10]
```diff
@@ -1,5 +1,7 @@
 const ApiEndpoints = {
-    Login_Endpoint: 'api/Auth/login'
+    Login_Endpoint: 'api/Auth/login',
+    User_Endpoint: (userId) => `api/User/${userId}`,
```
Comment (Catarina98) [inline GrocifyApp.UI/src/consts/ApiEndpoints.js]: remove User_Endpoint

**Review by Catarina98 — CHANGES_REQUESTED** [review]

File: GrocifyApp.API/Controllers/AuthController.cs (hunk 8 above)
Comment (jessywork) [reply GrocifyApp.API/Controllers/AuthController.cs]: we need userRequest to createToken

File: GrocifyApp.API/Models/RequestModels/UserRequestModel.cs (hunk 9 above)
Comment (jessywork) [reply GrocifyApp.API/Models/RequestModels/UserRequestModel.cs:16]: Does not isDarkMode need an example?

File: GrocifyApp.API/Controllers/UserController.cs (hunk 7 above)
Comment (jessywork) [reply GrocifyApp.API/Controllers/UserController.cs]: I don't understand what Authenticator.User is. I thought it was a variable with the data of the authenticator user.
However, User is the authenticated user and only has 3 claims. I added the email in order to call the api and get the user's details

File: GrocifyApp.API/Controllers/AuthController.cs (hunk 8 above)
Comment (Catarina98) [reply GrocifyApp.API/Controllers/AuthController.cs]: you can pass User to createToken

File: GrocifyApp.API/Models/RequestModels/UserRequestModel.cs (hunk 9 above)
Comment (Catarina98) [reply GrocifyApp.API/Models/RequestModels/UserRequestModel.cs:16]: remove isDarkMode from RequestModel, is never used

File: GrocifyApp.API/Controllers/UserController.cs (hunk 7 above)
Comment (Catarina98) [reply GrocifyApp.API/Controllers/UserController.cs]:
```cs
[HttpPut("toggleDarkMode")]
public async Task<ActionResult<User>> ToggleDarKMode()
{
    var currentUser = _mapper.Map<User>(AuthController.User);

    currentUser.IsDarkMode = !currentUser.IsDarkMode;

    try
    {
        await _userService.Update(currentUser);
    }
    catch (Exception ex)
    {
        var errors = new List<string> { ex.Message };

        return BadRequest(new BadResponseModel { Errors = errors });
    }

    var userResponseModel = _mapper.Map<UserResponseModel>(currentUser);

    return Ok(userResponseModel);
}
```

Why can't you use AuthController.User? and remove extra Claims

**Review by Catarina98 — APPROVED** [review]

**Pull request merged**